### ISSUE TITLE
Stability analysis missing exchange-correlation functional contributions

### ISSUE THREAD

**Issue by CDSherrill** [psi4/psi4#1045]
Currently, Psi4 runs a SCF stability analysis for DFT, even though contributions from the exchange-correlation functional derivatives have not yet been implemented, meaning the result is incorrect.  The driver should catch this and refuse to run the computation until it's fully implemented.  I assume this is a problem for both RKS and UKS, but @dgasmith can say if that's not correct.

**Comment by dgasmith** [comment]
We something like the following:

```
if (functional_->needs_xc()) {
    throw PSIEXCEPTION("Stability analysis not yet supported for XC functionals.");
}
```

At [this](https://github.com/psi4/psi4/blob/master/psi4/src/psi4/libscf_solver/uhf.cc#L903) location.

Great new dev project.

**Comment by dgasmith** [comment]
Closed via #1048.
